### PR TITLE
Fix adding selection path resulting in unneeded joins

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

If only the UUID or owned-by-id is required in a structural query, the table does not have to be joined.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1202805690238892/1203324626226299/f) _(internal)_

## 🔍 What does this change?

- Remove the `Option` fields in `EntityQueryPath`
- Avoid using a join on `EntityQueryPath::Uuid` and `EntityQueryPath::OwnedById`
- Replace `LEFT JOIN` with `INNER JOIN` again

### CODE REVIEW

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/entity.rs [hunk 1]
```diff
@@ -32,6 +32,11 @@ impl PostgresQueryRecord for EntityProperties {
 impl Path for EntityQueryPath<'_> {
     fn relations(&self) -> Vec<Relation> {
         match self {
+            Self::LeftEntity(path) | Self::RightEntity(path)
+                if **path == EntityQueryPath::Uuid || **path == EntityQueryPath::OwnedById =>
+            {
+                vec![]
+            }
```
Comment (TimDiekmann) [inline packages/graph/hash_graph/lib/graph/src/store/postgres/query/entity.rs:39]: It's possible to add more special cases here, like `UpdatedById`, do we want that?

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/statement/select.rs [hunk 2]
```diff
@@ -651,12 +651,8 @@ mod tests {
...
             SELECT *
             FROM "entities"
-            LEFT JOIN "entities" AS "entities_0_0_0"
-              ON "entities_0_0_0"."entity_uuid" = "entities"."left_entity_uuid"
-            LEFT JOIN "entities" AS "entities_0_0_1"
-              ON "entities_0_0_1"."entity_uuid" = "entities"."right_entity_uuid"
-            WHERE ("entities_0_0_0"."entity_uuid" = $1) AND ("entities_0_0_0"."owned_by_id" = $2)
-              AND ("entities_0_0_1"."entity_uuid" = $3) AND ("entities_0_0_1"."owned_by_id" = $4)
+            WHERE ("entities"."left_entity_uuid" = $1) AND ("entities"."left_owned_by_id" = $2)
+              AND ("entities"."right_entity_uuid" = $3) AND ("entities"."right_owned_by_id" = $4)
```
Comment (TimDiekmann) [inline packages/graph/hash_graph/lib/graph/src/store/postgres/query/statement/select.rs:655]: This is a really common query as it identifies a link by it's left/right entity. The join statement is now much shorter.

Comment (thehabbos007) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/query/statement/select.rs:655]: Yay! :sparkles:

**Review by thehabbos007 — APPROVED** [review]
Looks good to me! Just some comments for discussion, we can always address in follow-up if we merge now :)

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/entity.rs (hunk 1 above)
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/query/entity.rs:39]: This function will be removed in one of my next PR so I'd keep it as it is for now 🙂

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/expression/join_clause.rs [hunk 3]
```diff
@@ -25,7 +25,7 @@ impl<'q> JoinExpression<'q> {
 impl Transpile for JoinExpression<'_> {
     fn transpile(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
         // TODO: https://app.asana.com/0/1202805690238892/1203324626226299/f
-        fmt.write_str("LEFT JOIN ")?;
+        fmt.write_str("INNER JOIN ")?;
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/query/expression/join_clause.rs:28]: Yes exactly, this is the reason.

Let's hop on a call to further think about this later 🚀

**Comment by TimDiekmann** [comment]
Thanks, @thehabbos007!
2/3 of the concerns will be implicitly addressed in a follow-up. The third one *should* be fine, but I'm not 100% sure if a `NULL` join could still happen given a specific setup. I will test around a little bit more and report back.

**Comment by thehabbos007** [comment]
We discussed offline that it would make sense to differentiate between the join types, which we'll do in a follow-up

**Review by thehabbos007 — APPROVED** [review]

**Pull request merged**